src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.GssApiException.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class NetSecurityNative
+    {
+        internal sealed class GssApiException : Exception
+        {
+            private readonly Status _minorStatus;
+
+            public Status MinorStatus
+            {
+                get { return _minorStatus;  }
+            }
+
+            public GssApiException(string message) : base(message)
+            {
+            }
+
+            public GssApiException(Status majorStatus, Status minorStatus)
+                : base(GetGssApiDisplayStatus(majorStatus, minorStatus))
+            {
+                HResult = (int)majorStatus;
+                _minorStatus = minorStatus;
+            }
+
+            private static string GetGssApiDisplayStatus(Status majorStatus, Status minorStatus)
+            {
+                string majorError = GetGssApiDisplayStatus(majorStatus, isMinor: false);
+                string minorError = GetGssApiDisplayStatus(minorStatus, isMinor: true);
+
+                return (majorError != null && minorError != null) ?
+                    SR.Format(SR.net_gssapi_operation_failed_detailed, majorError, minorError) :
+                    SR.Format(SR.net_gssapi_operation_failed, majorStatus.ToString("x"), minorStatus.ToString("x"));
+            }
+
+            private static string GetGssApiDisplayStatus(Status status, bool isMinor)
+            {
+                GssBuffer displayBuffer = default(GssBuffer);
+
+                try
+                {
+                    Interop.NetSecurityNative.Status minStat;
+                    Interop.NetSecurityNative.Status displayCallStatus = DisplayStatus(out minStat, status, isMinor, ref displayBuffer);
```

Comment:
This passes `isMinor` to DisplayStatus.  DIsplayStatus calls that `isGssMechCode`; and it looks to me like that is used to determine which string table to load.

Are `isMinor` and `isGssMechCode` really the same?  And is that context specific for some reason?

My preference is to either expose the switch code from within the native function as a PAL enum, or break it into two separate methods; because either of those provides more clarity to a reader (and future developer) than `isGssMechCode` (since I have no idea what the implications of `false` are).